services/sharedchannel/service.go

```diff
@@ -109,8 +119,17 @@ func (scs *Service) Start() error {
 
 // Shutdown is called by the server on server shutdown.
 func (scs *Service) Shutdown() error {
+	rcs := scs.server.GetRemoteClusterService()
+	if rcs == nil {
+		return errors.New("Shared Channel Service cannot shutdown: requires Remote Cluster Service")
+	}
+
 	scs.mux.RLock()
 	id := scs.leaderListenerId
+	rcs.RemoveTopicListener(scs.syncTopicListenerId)
+	scs.syncTopicListenerId = ""
+	rcs.RemoveTopicListener(scs.inviteTopicListenerId)
+	scs.inviteTopicListenerId = ""
```

Comment:
If we do this we might want to use `Lock()` instead.

Reply:
Yup, good catch.  Lock is mandatory here.

Reply:
Fixed via https://github.com/mattermost/mattermost-server/pull/16880/commits/cc4bba95fe068054cc53451e3b2e533373bb1763